### PR TITLE
PE-98: Consider about how to restrict number of configs per set

### PR DESCRIPTION
**Pull request by dherdt**
### External References

* JIRA (Issue Tracker): https://issues.opennms.org/browse/PE-98

### CODE REVIEW

File: features/config/test/src/test/java/org/opennms/features/config/service/impl/ConfigurationManagerServiceImplTest.java [hunk 1]
```diff
@@ -179,31 +193,53 @@ public void testUpdateInvalidateConfiguration() throws Exception {
...
-        Optional<ConfigData<JSONObject>> configAfterDeregister = configManagerService.getConfigData(CONFIG_NAME);
+        configManagerService.unregisterSchema(CONFIG_NAME_MULTIPLE);
+        Optional<ConfigDefinition> schemaAfterDeregister = configManagerService.getRegisteredConfigDefinition(CONFIG_NAME_MULTIPLE);
+        Optional<ConfigData<JSONObject>> configAfterDeregister = configManagerService.getConfigData(CONFIG_NAME_MULTIPLE);
         Assert.assertTrue("FAIL TO deregister schema", schemaAfterDeregister.isEmpty());
         Assert.assertTrue("FAIL TO deregister config", configAfterDeregister.isEmpty());
     }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testErroneousConfideIdOnSchemaWithoutAllowMultiple() throws Exception {
```
Comment (patrick-schweizer) [inline features/config/test/src/test/java/org/opennms/features/config/service/impl/ConfigurationManagerServiceImplTest.java:225]: typo?

File: features/config/upgrade/src/main/resources/changelog-cm/30.0.0/changelog-cm.xml [hunk 2]
```diff
@@ -76,4 +76,11 @@
         </cm:changeSchema>
         <cm:importConfig schemaId="org.opennms.features.topology.app.icons.application" configId="default" filePath="org.opennms.features.topology.app.icons.application.cfg"/>
     </changeSet>
+
+    <!--TODO check which schemas can have multiple configurations-->
```
Comment (patrick-schweizer) [inline features/config/upgrade/src/main/resources/changelog-cm/30.0.0/changelog-cm.xml:80]: Open TODO?

**Review by patrick-schweizer — APPROVED** [review]
Looks good to me, just 2 minor comments

File: features/config/upgrade/src/main/resources/changelog-cm/30.0.0/changelog-cm.xml (hunk 2 above)
Comment (dherdt) [reply features/config/upgrade/src/main/resources/changelog-cm/30.0.0/changelog-cm.xml:80]: Ok. As as discussed I will consider all existing at this time configurations as non-multiple.

**Review by patrick-schweizer — DISMISSED** [review]

**Pull request closed without merging**